pkg/microscanner/scanner.go

```diff
@@ -52,8 +56,19 @@ func (s *scanner) scan(scanID uuid.UUID, req harbor.ScanRequest) error {
 		return fmt.Errorf("updating scan job status: %v", err)
 	}
 
+	dockerConfig, err := s.authorizer.Authorize(req)
+	if err != nil {
+		return fmt.Errorf("authorizing request: %v", err)
+	}
+	defer func() {
+		err := os.RemoveAll(dockerConfig)
```

Comment:
I think tmpDir is also unnecessary and should be removed.
```suggestion
		err := os.RemoveAll(filepath.Dir(dockerConfig))
```

Reply:
Good point. I've refactor slightly the code for handling docker config dir, but effectively I'm removing it on scan completion.